docs/source/known_issues.md

```diff
@@ -65,3 +65,44 @@ convert the datetimes to string.
 
 Timezone information is ignored at the moment, both when reading and when writing
 datetime columns.
+
+## Support for OpenStreetMap (OSM) data
+
+OpenStreetMap data do not natively support calculating the feature count by data
+layer due to the internal data structures. To get around this, Pyogrio iterates
+over all features first to calculate the feature count that is used to allocate
+arrays that contain the geometries and attributes read from the data layer, and
+then iterates over all feature again to populate those arrays. Further, data
+within the file are not structured at the top level to support fast reading by
+layer, which means that reading data by layer may need to read all records
+within the data source, not just those belonging to a particular layer. This is
+inefficient and slow, and is exacerbated when attemping to read from
+remotely-hosted data sources rather than local files.
+
+You may also be instructed by GDAL to enable interleaved reading mode via an
+error message when you try to read a large file without it, which you can do in
+one of two ways:
+
+1. Set config option used for all operations
+
+```python
+from pyogrio import set_gdal_config_options
+
+set_gdal_config_options({"OGR_INTERLEAVED_READING": True})
+```
+
+2. Set dataset open option
+
+```python
+
+from pyogrio import read_dataframe
+
+df = read_dataframe(path, INTERLEAVED_READING=True)
+```
+
+We recommend the following to sidestep performance issues:
+
```

Comment:
You could also mention use_arrow as an option. Based on my 1 test this is still not fast, but half as slow as without use_arrow=True. Or is it still too exprimental?

Reply:
I'm seeing different results for Alabama (https://download.geofabrik.de/north-america/us/alabama-latest.osm.pbf) reading the `lines` layer:

local, with arrow:  4.49s
local, without arrow: 3.46s
remote, with arrow: 23.55s
remote, without arrow: 20.26s

times are greater for `multipolygons` layer but follow similar trend locally but not for remote (likely because of much more network I/O, but it is remote enough from me that there could be huge variability in network I/O):

local, with arrow: 6.82s
local, without arrow: 5.50s
remote, with arrow: 33.85s
remote, without arrow: 39.06s

whereas for points, the first and fastest layer:
local, with arrow 1.12s
local, without arrow: 0.68s
remote, with arrow: 40.36s
remote, without arrow: 20.09s

I had expected the arrow option to be somewhat faster, but that doesn't seem to be the case for me.

Reply:
Hmm... my timings are indeed different:

``` python
from datetime import datetime
import pyogrio

url = "https://download.geofabrik.de/europe/germany/baden-wuerttemberg-latest.osm.pbf"
url = r"C:\temp\baden-wuerttemberg-latest.osm.pbf"

start = datetime.now()
pgdf = pyogrio.read_dataframe(url, use_arrow=False, layer="multipolygons", encoding="utf8")
print(f"took: {datetime.now() - start}")
```
Results:
```
# performance
# layer="multipolygons", use_arrow=True: 1:49
# layer="multipolygons", use_arrow=False: 3:47
```

PS: I had to specify `encoding="utf8"` to be able to run it with `use_arrow=False`

Reply:
PS: I had to specify `encoding="utf8"` to be able to run it with `use_arrow=False`. Based on the following post and others, it seems osm data is always in "UTF-8":
https://help.openstreetmap.org/questions/5308/character-encoding

So I think it would be better to default the `encoding` to "UTF-8" for .osm and .pbf?

Reply:
Added a fix to always set `UTF-8`.

It looks like using the interleaved encoding option has an impact on performance:

With default (no interleaved reading):
```
url=baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=True: 0:01:12.460372
url=baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=False: 0:03:11.110123
url=baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=True: 0:00:08.757978
url=baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=False: 0:00:15.286818
url=alabama-latest.osm.pbf, layer=multipolygons, use_arrow=True: 0:00:06.517222
url=alabama-latest.osm.pbf, layer=multipolygons, use_arrow=False: 0:00:15.456438
url=alabama-latest.osm.pbf, layer=points, use_arrow=True: 0:00:01.148906
url=alabama-latest.osm.pbf, layer=points, use_arrow=False: 0:00:02.335114
```

with `pyogrio.set_gdal_config_options({"OGR_INTERLEAVED_READING": True})`:

```
url=baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=True: 0:00:52.878035
url=baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=False: 0:01:05.556191
url=baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=True: 0:00:06.217804
url=baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=False: 0:00:04.641645
url=alabama-latest.osm.pbf, layer=multipolygons, use_arrow=True: 0:00:06.525850
url=alabama-latest.osm.pbf, layer=multipolygons, use_arrow=False: 0:00:05.703674
url=alabama-latest.osm.pbf, layer=points, use_arrow=True: 0:00:00.852143
url=alabama-latest.osm.pbf, layer=points, use_arrow=False: 0:00:00.724238
```

Reply:
Hmm... interleaved reading is consitently quite a bit faster... and often `use_arrow=False` becomes faster than `use_arrow=True` because of it. Kind of weird.

Reply:
On windows, a similar pattern:
```
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=True, interleaved=True: 0:01:49.743053
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=True, interleaved=False: 0:01:51.280849
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=False, interleaved=True: 0:01:28.155260
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=multipolygons, use_arrow=False, interleaved=False: 0:03:56.258161
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=True, interleaved=True: 0:00:20.615849
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=True, interleaved=False: 0:00:15.530900
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=False, interleaved=True: 0:00:10.094106
url=C:/Temp/baden-wuerttemberg-latest.osm.pbf, layer=points, use_arrow=False, interleaved=False: 0:00:28.546472
url=C:/Temp/alabama-latest.osm.pbf, layer=multipolygons, use_arrow=True, interleaved=True: 0:00:19.117897
url=C:/Temp/alabama-latest.osm.pbf, layer=multipolygons, use_arrow=True, interleaved=False: 0:00:19.194775
url=C:/Temp/alabama-latest.osm.pbf, layer=multipolygons, use_arrow=False, interleaved=True: 0:00:17.296830
url=C:/Temp/alabama-latest.osm.pbf, layer=multipolygons, use_arrow=False, interleaved=False: 0:00:35.047402
url=C:/Temp/alabama-latest.osm.pbf, layer=points, use_arrow=True, interleaved=True: 0:00:07.781301
url=C:/Temp/alabama-latest.osm.pbf, layer=points, use_arrow=True, interleaved=False: 0:00:07.374248
url=C:/Temp/alabama-latest.osm.pbf, layer=points, use_arrow=False, interleaved=True: 0:00:06.801308
url=C:/Temp/alabama-latest.osm.pbf, layer=points, use_arrow=False, interleaved=False: 0:00:09.139016
```

Reply:
I wonder if there is a reason why interleaved isn't the default...

Reply:
I am seeing the same behaviour. It's indeed a bit strange that with the interleaved reading turned on, reading with arrow is slower than without. Might be worth reporting to GDAL